File: LittleProjects/Scala/AkkaQuillSQLCrud/src/main/scala/akka/quill/sql/crud/repository/Repository.scala
package akka.quill.sql.crud.repository

import cats.data.OptionT
import cats.effect.IO

trait Repository[A, ID] {
  def findOne(id: ID): OptionT[IO, A]

  def findAll(): IO[Seq[A]]

  def save(a: A): IO[_]

  def update(a: A): IO[_]

  def deleteOne(id: ID): IO[_]

  def deleteAll(): IO[_]
}

File: LittleProjects/Java/GuiceHibernatePostgresCrud/src/main/java/guice/hibernate/postgres/crud/repository/Repository.java
package guice.hibernate.postgres.crud.repository;

import java.util.Optional;

public interface Repository<A, Id> {
    Optional<A> findOne(Id id);

    Iterable<A> findAll();

    A save(A a);

    A update(A a);

    boolean deleteOne(Id id);

    long deleteAll();
}
